File: PHP/uberBlack.php
<?php
    require_once('car.php');

    class UberBlack extends Car {
        public $typeCarAccepted;
        public $seatsMaterial;

        public function __construct($id, $license, $driver, $passegenger, $typeCarAccepted, $seatsMaterial) {
            parent::__construct($id, $license, $driver);
            $this->typeCarAccepted = $typeCarAccepted;
            $this->seatsMaterial = $seatsMaterial;
            $this->setPassegenger($passegenger);
        }

        public function __toString(){
            if ($this->passegenger != null) {
                return "[typeCarAccepted : " . print_r ($this->typeCarAccepted) . ", seatsMaterial : " . print_r ($this->seatsMaterial) . ", Car : " . parent::__toString() . "]";
            } else {
                return "";
            }  
        }

        public function setPassegenger($passegenger){
            if ($passegenger == 4) {
                $this->passegenger = $passegenger;
            } else {
                echo "Debe ingresar 4 passegenger";                
            }
        }
    }

?>

File: Java/UberBlack.java
import java.util.List;
import java.util.Map;

public class UberBlack extends Car{
    private Map<String, Map<String,Integer>> typeCarAccepted;
    private List<String> seatsMaterial;

    public UberBlack(Integer id, String license, Account driver, Integer passegenger,
    Map<String, Map<String,Integer>> typeCarAccepted, List<String> seatsMaterial) {
        super(id, license, driver);
        this.typeCarAccepted = typeCarAccepted;
        this.seatsMaterial = seatsMaterial;
        setPassegenger(passegenger);
    }

    public UberBlack(Integer id, String license, Account driver, Integer passegenger) {
        super(id, license, driver);
        setPassegenger(passegenger);
    }



    public Map<String, Map<String, Integer>> getTypeCarAccepted() {
        return typeCarAccepted;
    }

    public void setTypeCarAccepted(Map<String, Map<String, Integer>> typeCarAccepted) {
        this.typeCarAccepted = typeCarAccepted;
    }

    public List<String> getSeatsMaterial() {
        return seatsMaterial;
    }

    public void setSeatsMaterial(List<String> seatsMaterial) {
        this.seatsMaterial = seatsMaterial;
    }

    @Override
    public void setPassegenger(Integer passegenger) {
        if (passegenger != null && passegenger==4){
            super.setPassegenger(passegenger); 
        } else {
            System.out.println("Debe ingresar 4 passegenger");
        }
    }

    @Override
    public String toString() {
        if (super.getPassegenger() != null) {
            return "UberBlack [seatsMaterial=" + seatsMaterial + ", typeCarAccepted=" + typeCarAccepted + ", " + super.toString()+ "]";
        }
        return "";
    }

}
